fixups/FileRedirectionFixup/PathRedirection.cpp

```diff
@@ -532,54 +612,129 @@ std::wstring UrlDecode(std::wstring str)
         }
         else 
         {
-            swscanf_s(str.substr(i + 1, 2).c_str(), L"%x", &ii);
-            ch = static_cast<char>(ii);
-            ret += ch;
-            i = i + 2;
+            if (swscanf_s(str.substr(i + 1, 4).c_str(), L"%x", &ii) == 1)
+            {
+                ch = static_cast<char>(ii);
+                ret += ch;
+                if (ii <= 255)
+                    i = i + 2;
+                else
+                    i = i + 4;
+            }
+            else
+            {
+                ret += str[i];
+            }
         }
     }
     return ret;
 }
 
 std::string StripFileColonSlash(std::string old_string)
 {
-    size_t found = old_string.find("file:\\",0);
-    if (found != 0)
-    {
-        found = old_string.find("file:/", 0);
-    }
-    if (found != 0)
+    int count = 0;
+    size_t found = old_string.find("file:\\\\",0);
+    if (found == 0)
     {
-        found = old_string.find("FILE:\\", 0);
+        count = 7;

```

Comment:
Can you please replace the magic number 7 with a constant so we can have a way to tell what this number is for.

Reply:
Addressed in Commit 1bdc01a3

Reply:
Noticed that I missed the wstring version on this.  Updated both string and wstring versions, possibly with some simpler code.